frontend/src/utils/toolErrorHandler.ts

```diff
@@ -2,16 +2,72 @@
  * Standardized error handling utilities for tool operations
  */
 
+/**
+ * Standard error type that covers common error patterns
+ */
+export interface ToolError {
+  message?: string;
+  response?: {
+    data?: string | unknown;
+    status?: number;
+  };
+}
+
+/**
+ * Extract error message from JSON response data
+ */
+const extractFromJsonData = (data: unknown): string | null => {
+  if (typeof data === 'object' && data !== null) {
+    const obj = data as Record<string, unknown>;
+    // Common JSON error patterns
+    if (typeof obj.message === 'string' && obj.message.trim()) {
+      return obj.message.trim();
+    }
+    if (typeof obj.error === 'string' && obj.error.trim()) {
+      return obj.error.trim();
+    }
+    if (typeof obj.detail === 'string' && obj.detail.trim()) {
+      return obj.detail.trim();
+    }
+  }
+  return null;
+};
+
 /**
  * Default error extractor that follows the standard pattern
  */
-export const extractErrorMessage = (error: any): string => {
-  if (error.response?.data && typeof error.response.data === 'string') {
-    return error.response.data;
+export const extractErrorMessage = (error: unknown): string => {
+  const typedError = error as ToolError;
```

Comment:
```suggestion
export const extractErrorMessage = (error: ToolError): string => {
```

Making the param type into `unknown` and then casting it on the first line means that call-sites can pass anything into it and then the type error is silenced for the rest of the function. We should either be requiring the call-sites to be passing in things that meet the interface, or have an early return which checks that the passed object does meet the interface rather than blindly casting it.